planner/core/exhaust_physical_plans.go

```diff
@@ -2259,9 +2312,16 @@ func (la *LogicalAggregation) getHashAggs(prop *property.PhysicalProperty) []Phy
 		taskTypes = []property.TaskType{property.RootTaskType}
 	}
 	for _, taskTp := range taskTypes {
-		agg := NewPhysicalHashAgg(la, la.stats.ScaleByExpectCnt(prop.ExpectedCnt), &property.PhysicalProperty{ExpectedCnt: math.MaxFloat64, TaskTp: taskTp})
-		agg.SetSchema(la.schema.Clone())
-		hashAggs = append(hashAggs, agg)
+		if taskTp == property.MppTaskType {
```

Comment:
As I have mentioned, there is no need to append mpp task to all task types, so that we can move this exception out of the loop, to avoid **messing up** the original code.

Reply:
in addition to inserting the MppTaskType here at [Line ](https://github.com/pingcap/tidb/pull/21842/files#diff-d3d0144d9fe67ce1ca77fb02c6ebb7895cade0809e47b21b926bfa29feb0efb2R2297),
the MppTaskType may also come from prop at [Line](https://github.com/pingcap/tidb/pull/21842/files#diff-d3d0144d9fe67ce1ca77fb02c6ebb7895cade0809e47b21b926bfa29feb0efb2R2309) , so we should specifically process the MppTaskType at the loop.

Reply:
I don't catch it. MppTask can come from where?

Reply:
update the link. sorry it became invalid after updated.